catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -151,143 +162,134 @@ public void setAttributeOverrides(List<String> attributeOverrides) {
+    }
 
-        RouteBuilder routeBuilder = new RouteBuilder() {
-            @Override
-            public void configure() throws Exception {
-                String inbox = "file:" + monitoredDirectory + "?moveFailed=.errors";
-                if (copyIngestedFiles) {
-                    inbox += "&move=.ingested";
-                } else {
-                    inbox += "&delete=true";
-                }
-                LOGGER.debug("inbox = {}", inbox);
-
-                RouteDefinition routeDefinition = from(inbox);
-
-                if (attributeOverrides != null) {
-                    String attributeOverrideString = attributeOverrides.stream()
-                            .map(String::trim)
-                            .collect(Collectors.joining(","));
-                    routeDefinition.setHeader(Constants.ATTRIBUTE_OVERRIDES_KEY,
-                            simple(attributeOverrideString));
-                }
-
-                routeDefinition.process(systemSubjectBinder)
-                        .to("content:framework");
-            }
-        };
-
+    /*
+        The task method used to verify the "content" CamelComponent before loading routes.
+     */
+    private void attemptAddRoutes() {
+        LOGGER.debug("Attempting to add routes for CDM {}", runtimeMonitorId);
+        RouteBuilder routeBuilder = createRouteBuilder();
         try {
-            // Add the routes that will be built by the RouteBuilder class above
-            // to this CamelContext.
-            // The addRoutes() method will instantiate the RouteBuilder class above,
-            // and start the routes (only) if the camelContext has already been started.
+            waitForContentCamelComponent();
             camelContext.addRoutes(routeBuilder);
-
-            // Save the routes created by RouteBuilder so that they can be
-            // stopped and removed later if the route(s) are modified by the
-            // administrator or this ContentDirectoryMonitor is deleted.
-            this.routeCollection = routeBuilder.getRouteCollection()
-                    .getRoutes();
-
-            // Start route that was just added.
-            // If the route was just added for the first time, i.e., this not a bundle
-            // restart, then this method will do nothing since the addRoutes() above
-            // already started the route. But for bundle (or system) restart this call
-            // is needed since the addRoutes() for whatever reason did not start the route.
-            startRoutes();
-
+            setRouteCollection(routeBuilder);
+        } catch (InterruptedException e) {
+            Thread.currentThread()
+                    .interrupt();
+            LOGGER.warn("CDM interrupted while waiting for Camel component 'content'. ");
+        } catch (Exception e) {
+            LOGGER.warn(
+                    "Unable to configure Camel route - this Content Directory Monitor will be unusable");
```

Comment:
I would not log on each failure but only log if all attempts fail.  We do not want to see multiple exceptions that are then followed by a success since that would be misleading.